pkg/virt-handler/migration-source.go

```diff
@@ -484,9 +484,19 @@ func (c *MigrationSourceController) migrateVMI(vmi *v1.VirtualMachineInstance, d
 		return fmt.Errorf("failed to handle migration proxy: %v", err)
 	}
 
-	migrationConfiguration := vmi.Status.MigrationState.MigrationConfiguration
-	if migrationConfiguration == nil {
+	var migrationConfiguration *v1.MigrationConfiguration
+	if vmi.Status.MigrationState.MigrationConfiguration == nil {
 		migrationConfiguration = c.clusterConfig.GetMigrationConfiguration()
+	} else {
+		migrationConfiguration = vmi.Status.MigrationState.MigrationConfiguration.DeepCopy()
+	}
+
+	// This check is only for backward compatibility.
+	// During upgrade, AllowWorkloadDisruption could be nil since the migration controller is
+	// updated later the virt-handler.
+	// This check can be removed in future
+	if migrationConfiguration.AllowWorkloadDisruption == nil {
+		migrationConfiguration.AllowWorkloadDisruption = migrationConfiguration.AllowPostCopy
```

Comment:
This is not something we want, `AllowPostCopy` should be ignored when `AllowWorkloadDisruption` isn't set. Why can't we just add a nil pointer check wherever the nil pointer exception is happening instead?